docs/design/NsfsVersioning.md

```diff
@@ -49,6 +49,9 @@ In the figure, each original directory contains a hidden .versions/ sub- directo
 * The version ID will be attached as an xattr of the file.
 * When a latest version is moved to past versions directory, the file name will be changed from key to key_{version_id}.
 
+#### Multipart upload version order
+according to AWS specifications multipart upload version time should be calcluated based on multipart upload creation. see https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html#distributedmpupload time rather then completion time. By design on our system the version-id time is calculated based on completion time.
```

Comment:
Don't forget to add here a disclaimer about the directory buckets so it's not exactly not following AWS S3 protocol
same in the comment in namespace_fs

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-using-multipart-upload.html
For S3 Express One Zone, the object creation time is the completion date of the multipart upload.